### PR TITLE
wallet: fix ghost UTXOs not being detected

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightningnetwork/lnd/issues/6434 by bumping the `btcwallet` version to the latest version that includes the fix in https://github.com/btcsuite/btcwallet/pull/838.

We also add an integration test that was able to show the problem.

Fixes https://github.com/lightningnetwork/lnd/issues/6434

### CODE REVIEW

File: lntest/itest/lnd_recovery_test.go [hunk 1]
```diff
@@ -282,3 +289,156 @@ func testOnchainFundRecovery(ht *lntemp.HarnessTest) {
...
+	// (m/1017') key scope. Because the internal key scope is defined as
+	// p2wkh address type, this might be incorrectly detected by the wallet
+	// in some situations (which this test makes sure is fixed).
+	const ghostUtxoAmount = 456_000
+	carol.RPC.SendCoins(&lnrpc.SendCoinsRequest{
+		Addr:        ghostUtxoAddr.String(),
+		Amount:      ghostUtxoAmount,
+		SatPerVbyte: 1,
+	})
+	ht.Miner.AssertNumTxsInMempool(1)
```
Comment (yyforyongyu) [inline lntest/itest/lnd_recovery_test.go:352]: nit: could use `ht.MineBlocksAndAssertNumTxes(1)` instead, which checks the mempool and mine blocks.

File: lntest/itest/lnd_recovery_test.go [hunk 2]
```diff
@@ -282,3 +289,156 @@ func testOnchainFundRecovery(ht *lntemp.HarnessTest) {
...
+	ht.MineBlocks(1)
+
+	// Make sure we see the change output in our list of unspent outputs.
+	// We _don't_ expect to see the ghost UTXO here as in this step it's
+	// ignored as an internal address correctly.
+	ht.AssertNumUTXOsConfirmed(carol, 1)
+	unspent := carol.RPC.ListUnspent(&walletrpc.ListUnspentRequest{
+		MinConfs: 1,
+	})
+	require.Len(ht, unspent.Utxos, 1)
```
Comment (yyforyongyu) [inline lntest/itest/lnd_recovery_test.go:362]: hmm I think it's already check by `AssertNumUTXOsConfirmed`...

File: lntest/itest/lnd_recovery_test.go (hunk 1 above)
Comment (guggero) [reply lntest/itest/lnd_recovery_test.go]: Ah, wasn't aware that existed, but good to know. Changed it to that call.

File: lntest/itest/lnd_recovery_test.go [hunk 3]
```diff
@@ -282,3 +289,158 @@ func testOnchainFundRecovery(ht *lntemp.HarnessTest) {
 
 	restoreCheckBalance(finalBalance-minerAmt-fee, 1, 21, nil)
 }
+
+// testRescanAddressDetection makes sure that addresses created from internal
+// (m/1017' scope) keys aren't detected as UTXOs when re-scanning the wallet
+// with --reset-wallet-transactions to avoid showing them as un-spent ghost
+// UTXOs even if they are being spent. This is to test a fix in the wallet that
+// addresses the following scenario:
```
Comment (Roasbeef) [inline lntest/itest/lnd_recovery_test.go:297]: Excellent reproduction 👍

File: lntest/itest/lnd_recovery_test.go [hunk 4]
```diff
@@ -282,3 +289,158 @@ func testOnchainFundRecovery(ht *lntemp.HarnessTest) {
...
+	require.NoError(ht, tx.Serialize(&buf))
+
+	ghostUtxoScript := ht.PayToAddrScript(ghostUtxoAddr)
+	utxoInfo := []*signrpc.TxOut{{
+		PkScript: ghostUtxoScript,
+		Value:    ghostUtxoAmount,
+	}}
+
+	// Let's sign the input now.
+	signResp := carol.RPC.SignOutputRaw(&signrpc.SignReq{
```
Comment (Roasbeef) [inline lntest/itest/lnd_recovery_test.go:406]: Also love how compact itests can be now that the new framework has gotten rid of a lot of the exiting boiler plate.

**Review by Roasbeef — APPROVED** [review]
LGTM 🌻

btcwallet dep has been meged, so the `go.mod` here can be updated.

**Review by positiveblue — APPROVED** [review]
Perfect, i was waiting for the btc dependency to be updated to approve :+1: 

It looks like the lint is complaining though (and the code format too)

**Comment by Roasbeef** [comment]
Linter failing: 
```
Error: lntemp/harness_assertion.go:1684:9: ineffectual assignment to err (ineffassign)
		addr, err = btcutil.NewAddressScriptHash(
```

**Comment by guggero** [comment]
Fixed the linter, but the etcd/postgres itests also seem to be failing. Will investigate tomorrow.

**Comment by guggero** [comment]
I found the issue. The `--reset-wallet-transactions` flag didn't work at all with remote databases such as etcd or Postgres because of an edge case in the wallet loader. I added a commit to re-initialize the loader as well to fix the problem.

**Review by yyforyongyu — APPROVED** [review]

**Pull request merged**